django/contrib/admin/options.py

```diff
@@ -1522,7 +1522,7 @@ def changelist_view(self, request, extra_context=None):
         if (request.method == "POST" and cl.list_editable and
                 '_save' in request.POST and not action_failed):
             FormSet = self.get_changelist_formset(request)
-            formset = cl.formset = FormSet(request.POST, request.FILES, queryset=cl.result_list)
+            formset = cl.formset = FormSet(request.POST, request.FILES, queryset=self.get_queryset(request))
```

Comment:
Could also use ``cl.get_queryset(request)`` here, but the filtering and ordering imposed by the Changelist method might be at least in part responsible for some of the other behaviour people were reporting in the ticket.